### PR TITLE
Update liblinear

### PR DESCRIPTION
**Pull request by ericphanson**
So it builds for m1, but can also bump the version at the same time

### CODE REVIEW

**Comment by ericphanson** [comment]
The patch seems to be updating the extensions in the Makefile, so I've tried to make the same adjustments to the [v247 makefile](https://github.com/cjlin1/liblinear/blob/v247/Makefile) in the updated patch here.

File: L/liblinear/build_tarballs.jl [hunk 1]
```diff
@@ -17,7 +17,7 @@ cd $WORKSPACE/srcdir
 for f in ${WORKSPACE}/srcdir/patches/*.patch; do
     atomic_patch -p1 ${f}
 done
-cd liblinear-230/
+cd liblinear/
 mkdir -p ${prefix}/bin
 mkdir -p ${prefix}/lib
 if [[ "${target}" == *-freebsd* ]] || [[ "${target}" == *-apple-* ]]; then
```
Comment (giordano) [inline L/liblinear/build_tarballs.jl:23]: Uhm, I don't remember why were using gcc, it'd be good to see if we can avoid that

File: L/liblinear/build_tarballs.jl [hunk 2]
```diff
@@ -17,7 +17,7 @@ cd $WORKSPACE/srcdir
 for f in ${WORKSPACE}/srcdir/patches/*.patch; do
     atomic_patch -p1 ${f}
 done
-cd liblinear-230/
+cd liblinear/
 mkdir -p ${prefix}/bin
```
Comment (giordano) [inline L/liblinear/build_tarballs.jl:21]:
```suggestion
mkdir -p "${bindir}"
```

File: L/liblinear/build_tarballs.jl [hunk 3]
```diff
@@ -17,7 +17,7 @@ cd $WORKSPACE/srcdir
...
 mkdir -p ${prefix}/lib
```
Comment (giordano) [inline L/liblinear/build_tarballs.jl:22]:
```suggestion
mkdir -p "${libdir}"
```

File: L/liblinear/build_tarballs.jl (hunk 1 above)
Comment (ericphanson) [reply L/liblinear/build_tarballs.jl:23]: should I try just deleting this `if` block?

Comment (ericphanson) [reply L/liblinear/build_tarballs.jl]: trying that in https://github.com/JuliaPackaging/Yggdrasil/pull/7698/commits/724addf649f26a9cea343551d1e0b83c0b17e0b8

**Comment by ericphanson** [comment]
success!

**Pull request merged**